presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -474,7 +474,14 @@ public StreamProperties visitTopN(TopNNode node, List<StreamProperties> inputPro
         @Override
         public StreamProperties visitSort(SortNode node, List<StreamProperties> inputProperties)
         {
-            return StreamProperties.ordered();
+            StreamProperties sourceProperties = Iterables.getOnlyElement(inputProperties);
+            if (sourceProperties.isSingleStream()) {
```

Comment:
maybe it should be in a `checkState` as currently it would mean that there is a bug somewhere, no point of having parallel sort without a merge